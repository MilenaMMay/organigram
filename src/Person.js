import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { container, heading, subline } from './Person.styles'

export default class Person extends Component {
  static propTypes = {
    person: PropTypes.object,
  }

  render() {
    const {
      person,
    } = this.props

    return (
      <div style={container}>
        <h1 style={heading}>{person.firstname} {person.lastname}</h1>
        <h2 style={subline}>{person.role}</h2>
      </div>
    )
  }
}
