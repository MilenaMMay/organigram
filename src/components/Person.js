import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Card from '@material-ui/core/Card'

import { container, heading, subline } from '../styles'

export default class Person extends Component {
  static propTypes = {
    person: PropTypes.object,
  }

  render() {
    const {
      person,
    } = this.props

    return (
      <div style={container}>
        <h1 style={heading}>{person.firstname} {person.lastname}</h1>
        <h2 style={subline}>{person.role}</h2>
      </div>
    )
  }
}
