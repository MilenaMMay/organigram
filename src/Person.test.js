import React from 'react';
import Person from './Person'
import renderer from 'react-test-renderer';

test('Person renders views', () => {
  const component = renderer.create(
    <Person person={{ firstname: 'Foo', lastname: 'Bar', role: 'foobar' }}></Person>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
