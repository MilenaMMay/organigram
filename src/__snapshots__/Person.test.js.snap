// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person renders views 1`] = `
<div
  style={
    Object {
      "border": "2px solid #000",
      "display": "inline-block",
      "margin": "2em",
      "padding": "1em",
    }
  }
>
  <h1
    style={
      Object {
        "font-size": "1.5em",
      }
    }
  >
    Foo
     
    Bar
  </h1>
  <h2
    style={
      Object {
        "font-size": "1em",
      }
    }
  >
    foobar
  </h2>
</div>
`;
